Modules/Loadable/Markups/MRMLDM/vtkMRMLMarkupsDisplayableManager.cxx

```diff
@@ -561,51 +586,37 @@ bool vtkMRMLMarkupsDisplayableManager::IsCorrectDisplayableManager()
 //---------------------------------------------------------------------------
 bool vtkMRMLMarkupsDisplayableManager::IsManageable(vtkMRMLNode* node)
 {
-  return (this->Focus.find(node->GetClassName()) != this->Focus.end());
+  return (this->MarkupsNodesWidgets.find(node->GetClassName()) != this->MarkupsNodesWidgets.end());
 }
 
 //---------------------------------------------------------------------------
 bool vtkMRMLMarkupsDisplayableManager::IsManageable(const char* nodeClassName)
 {
-  return nodeClassName && (this->Focus.find(nodeClassName) != this->Focus.end());
+  return nodeClassName && (this->MarkupsNodesWidgets.find(nodeClassName) != this->MarkupsNodesWidgets.end());
 }
 
 //---------------------------------------------------------------------------
 vtkMRMLMarkupsNode* vtkMRMLMarkupsDisplayableManager::CreateNewMarkupsNode(
   const std::string &markupsNodeClassName)
 {
-  // create the MRML node
   std::string nodeName = "M";
-  if (markupsNodeClassName == "vtkMRMLMarkupsFiducialNode")
-    {
-    nodeName = "F";
-    }
-  else if (markupsNodeClassName == "vtkMRMLMarkupsAngleNode")
-    {
-    nodeName = "A";
-    }
-  else if (markupsNodeClassName == "vtkMRMLMarkupsLineNode")
-    {
-    nodeName = "L";
-    }
-  else if (markupsNodeClassName == "vtkMRMLMarkupsCurveNode")
-    {
-    nodeName = "C";
-    }
-  else if (markupsNodeClassName == "vtkMRMLMarkupsClosedCurveNode")
-    {
-    nodeName = "C";
-    }
-  else if (markupsNodeClassName == "vtkMRMLMarkupsPlaneNode")
+
+  for (const auto &item : this->MarkupsNames)
```

Comment:
Why not use find_if with a lambda?  I know the loop is sure to be short but I think it would be a good pattern.

Reply:
Actually, this should be just a method call to the widget object